File: SmartValley.WebApi/Projects/Responses/ProjectResponse.cs
using SmartValley.Domain;
using SmartValley.Domain.Entities;

namespace SmartValley.WebApi.Projects.Responses
{
    public class ProjectResponse
    {
        public long Id { get; set; }

        public string Name { get; set; }

        public string Country { get; set; }

        public Category Category { get; set; }

        public ProjectScoringResponse Scoring { get; set; }

        public string Description { get; set; }

        public bool IsApplicationSubmitted { get; set; }

        public ScoringStartTransactionStatus ScoringStartTransactionStatus { get; set; }

        public string ScoringStartTransactionHash { get; set; }

        public static ProjectResponse Create(Project project, ScoringApplication scoringApplication)
        {
            return new ProjectResponse
                   {
                       Id = project.Id,
                       Name = project.Name,
                       Country = project.Country?.Code,
                       Category = project.Category,
                       Description = project.Description,
                       IsApplicationSubmitted = scoringApplication?.IsSubmitted ?? false,
                       ScoringStartTransactionStatus = scoringApplication?.GetTransactionStatus() ?? ScoringStartTransactionStatus.NotSubmitted,
                       ScoringStartTransactionHash = scoringApplication?.ScoringStartTransaction?.Hash,
                       Scoring = project.Scoring == null ? null : ProjectScoringResponse.Create(project.Scoring)
                   };
        }
    }
}

File: SmartValley.FrontEnd/src/app/api/project/project-response.ts
import {Category} from '../../services/common/category';
import {ProjectScoringResponse} from './project-scoring-response';
import {ScoringStartTransactionStatus} from './scoring-start-transaction.status';

export interface ProjectResponse {
  id: number;
  imageUrl: string;
  name: string;
  country: string;
  category: Category;
  description: string;
  scoring?: ProjectScoringResponse;
  isApplicationSubmitted: boolean;
  scoringStartTransactionStatus: ScoringStartTransactionStatus;
  scoringStartTransactionHash: string;
  authorAddress?: string;
}
